### PR TITLE
Start the Grandmapocalypse

### PR DESCRIPTION
**Pull request by fejta**
This sticks an access token for `service-account` in `http-cookiefile`

Will rewrite as a golang program shortly

### CODE REVIEW

File: prow/cmd/grandmatriarch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+FROM gcr.io/cloud-builders/gcloud
+LABEL maintainer="fejta@google.com"
+
+COPY fetch.sh /fetch.sh
```
Comment (krzyzacy) [inline prow/cmd/grandmatriarch/Dockerfile]: bake.sh?

File: prow/cmd/grandmatriarch/bake.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# TODO(fejta): make this a good program, not bash
```
Comment (krzyzacy) [inline prow/cmd/grandmatriarch/bake.sh]: :joy:

File: prow/cluster/BUILD.bazel [hunk 3]
```diff
@@ -11,6 +11,7 @@ release(
     component("deck", "service", "deployment"),
     component("gce-ssd-retain", "storageclass"),
     component("ghproxy", "service", "deployment"),
+    component("grandmatriarch", MULTI_KIND, cluster=BUILD_CLUSTER),
```
Comment (krzyzacy) [inline prow/cluster/BUILD.bazel]: do we want to add it to our prow deployment?

File: prow/cmd/grandmatriarch/bake.sh [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  gcloud config config-helper --force-auth-refresh | grep access_token | grep -o -E '[^ ]+$'
+}
+
+# DOMAIN
+# INITIAL_DOT
+# PATH
+# PATH_SPECIFIED
+# expires
+# name
+# value
```
Comment (krzyzacy) [inline prow/cmd/grandmatriarch/bake.sh]: what's this?

File: prow/cluster/BUILD.bazel (hunk 3 above)
Comment (fejta) [reply prow/cluster/BUILD.bazel]: That is what this does?

File: prow/cmd/grandmatriarch/Dockerfile (hunk 1 above)
Comment (fejta) [reply prow/cmd/grandmatriarch/Dockerfile]: Done

File: prow/cmd/grandmatriarch/bake.sh (hunk 4 above)
Comment (fejta) [reply prow/cmd/grandmatriarch/bake.sh]: Done

File: prow/cmd/grandmatriarch/bake.sh (hunk 2 above)
Comment (fejta) [reply prow/cmd/grandmatriarch/bake.sh]: Ack

File: prow/cmd/grandmatriarch/bake.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  if [[ "$#" != 4 ]]; then
+    echo "Usage: print-cookie <HOST> <IS_DOT> <EXPIRES_EPOCH> <TOKEN>" >&2
+    return 1
+  fi
+  host="$1"
+  dot="$2"
+  exp="$3"
+  tok="$4"
+  for part in "$host" "$dot" / TRUE "$exp" o; do
+    echo -n ${part}$'\t' # apparently $'\t' is tab
```
Comment (BenTheElder) [inline prow/cmd/grandmatriarch/bake.sh]: https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html works for ANSI C strings in general, fwiw

File: prow/cluster/BUILD.bazel (hunk 3 above)
Comment (krzyzacy) [reply prow/cluster/BUILD.bazel]: (I mean we want to add it to the gerrit deployment, not k8s one? or given nobody is using the token so this is fine I guess)

Comment (fejta) [reply prow/cluster/BUILD.bazel]: I think this is okay in the k8s cluster?

**Pull request merged**